File: dayTwo/Java/TestBookSystem.java
// PSEUDOCODE:
// Test 1: Can we add a book?
// Test 2: Does add return success message?
// Test 3: Can we remove a book?
// Test 4: Can we update a book title?

import org.junit.jupiter.api.*;
import static org.junit.jupiter.api.Assertions.*;

public class TestBookSystem {
    
    @BeforeEach
    void cleanBooksBeforeEachTest() {
        BookSystem.bookStorage = new String[10];
        BookSystem.numberOfBooks = 0;
    }
    
    @Test
    void testThatAddBookIncreasesBookCount() {
        BookSystem.addBook("The Hobbit");
        assertEquals(1, BookSystem.numberOfBooks);
    }
    
    @Test
    void testThatAddBookReturnsSuccessMessage() {
        String result = BookSystem.addBook("Animal Farm");
        assertEquals("Book added successfully", result);
    }
    
    @Test
    void testThatAddingSameBookReturnsErrorMessage() {
        BookSystem.addBook("The Mystery");
        String result = BookSystem.addBook("The Mystery");
        assertEquals("Book already exists", result);
    }
    
    @Test
    void testThatRemoveBookDecreasesBookCount() {
        BookSystem.addBook("Harry Potter");
        BookSystem.removeBook("Harry Potter");
        assertEquals(0, BookSystem.numberOfBooks);
    }
    
    @Test
    void testThatUpdateBookChangesTitle() {
        BookSystem.addBook("Brave Kind");
        BookSystem.updateBook("Brave Kind", "Brave Kingdom");
        String[] allBooks = BookSystem.showAllBooks();
        assertEquals("Brave Kingdom", allBooks[0]);
    }
}

File: dayTwo/Python/BookSystem/test_book_system.py
# PSEUDOCODE:
# Test if we can add a book
# Test if we can remove a book  
# Test if we can update a book
# Test if we can see all books
# Test if we can get a random book suggestion

import unittest
import book_system

class TestBookSystem(unittest.TestCase):
    
    def test_that_add_book_works(self):
        book_system.add_book("The Hobbit")
        
    def test_that_add_book_returns_success(self):
        result = book_system.add_book("The Mystery")
        self.assertEqual(result, "Book added successfully")
    
    def test_that_add_same_book_returns_error(self):
        book_system.add_book("The Mystery")
        result = book_system.add_book("The Mystery")
        self.assertEqual(result, "Book already exists")
    
    def test_that_remove_book_works(self):
        book_system.add_book("Harry Potter")
        book_system.remove_book("Harry Potter")
        
    def test_that_remove_book_returns_success(self):
        book_system.add_book("1984")
        result = book_system.remove_book("1984")
        self.assertEqual(result, "Book removed successfully")
    
    def test_that_update_book_works(self):
        book_system.add_book("Brave Kind")
        book_system.update_book("Brave Kind", "Brave Kingdom")
    
    def test_that_suggest_book_returns_two_things(self):
        book_system.add_book("The Hobbit")
        book, page = book_system.suggest_book()
        self.assertEqual(book, "The Mystery")
